examples/controls_example/public/app/react_control_example/react_control_example.tsx

```diff
@@ -361,9 +363,15 @@ export const ReactControlExample = ({
             </EuiFlexItem>
             <EuiFlexItem grow={false}>
               <EuiButtonEmpty
-                isDisabled={!controlGroupApi}
-                onClick={() => {
-                  controlGroupApi?.resetUnsavedChanges();
+                isDisabled={!controlGroupApi || isResetting}
+                isLoading={isResetting}
+                onClick={async () => {
+                  if (!controlGroupApi) {
+                    return;
+                  }
+                  setIsResetting(true);
+                  await controlGroupApi.asyncResetUnsavedChanges();
+                  if (isMounted()) setIsResetting(false);
```

Comment:
After resetting a second time, `controlGroupApi.asyncResetUnsavedChanges` doesn't ever seem to resolve:

https://github.com/user-attachments/assets/449cea12-824f-4091-afc5-d172c71ae0e8

Reply:
resolved with b78a3014e8edd12a39b31c85300fe4f8beac8729.

Search control had a `distinctUntilChanged` blocking `setOutputFilters` from being called. Removing this removed the problem.